clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -29,15 +33,25 @@
      * @param topic The topic this record is received from
      * @param partition The partition of the topic this record is received from
      * @param offset The offset of this record in the corresponding Kafka partition
+     * @param timestamp The timestamp of the record.
+     * @param timestampType The timestamp type
      * @param key The key of the record, if one exists (null is allowed)
      * @param value The record contents
      */
-    public ConsumerRecord(String topic, int partition, long offset, K key, V value) {
+    public ConsumerRecord(String topic,
```

Comment:
how would you feel about adding a testing constructor for this class that matches the existing signature, and would just use dummy values for the timestamp and `TimestampType`?  Eliminating that constructor will break compilation for any unit tests that rely on building `ConsumerRecord`s